includes/class-sensei-notices.php

```diff
@@ -106,8 +108,50 @@ public function maybe_print_notices() {
 
 		// set this to print immediately if notices are added after the notices were printed.
 		$this->has_printed = true;
+	}
 
-	} // end print_notice()
+	/**
+	 * Adds a filter to the content to add notices added by blocks.
+	 */
+	public function setup_block_notices() {
+		if ( is_singular( 'course' ) && ! Sensei()->course->is_legacy_course( get_post() ) ) {
```

Comment:
I wonder if we could remove the template actions where the notices are added, and add them generically to the whole application in `the_content`, as you did here. Does that make sense? Or we could have some issue doing that?

Reply:
What we trying to avoid is notices being printed erroneously in old, non block-based posts.

As we might look into using `Sensei_Notices` in many post types and pages in the future, each post type/page might have a different method to check if the page has blocks in it so we need to perform the check after `template_redirect`.